### ISSUE TITLE
Opening .sav file, graphics output

### ISSUE THREAD

**Issue by baltzis** [jasp-stats/jasp-desktop#1695]
## Environment (for bugs)

* JASP version: 0.8.1.1
* OS name and version: Windows 10 (64bit)
* Analysis: When I try to open a .sav file in a directory other than route, I get the message:
"Unable to open file. No header found in .SAV file".
I suspect that this is because my system is in Greek and several directories have Greek names. The same file, however, can be opened from the route directory of "My Documents".

* Second problem: While labels in Greek appear OK in tables (e.g. frequencies), in plots they appear in gibberish (see attached image).
![test-jasp](https://cloud.githubusercontent.com/assets/26595356/24191178/a3b22c2e-0ef3-11e7-8830-aa1e5c0dbff7.jpg)

**Comment by jdittrich** [comment]
It seems that this issue should be split in two issues, so that both issues could be investigated and solved independently

e.g. in these: 
* "Unable to open file. No header found in .SAV file when in folder with Greek characters"
* "Greek characters are correct in tables but wrongly encoded in graphs"

**Comment by AlexanderLyNL** [comment]
The plot problem is indeed an R problem. At the moment, we cannot have foreign languages in the plots.

**Comment by aknight1-uva** [comment]
The "No Header issue" strikes me as odd. - It seems like a file has been found (and opened) but the contents of the file are not recognisable as an .SAV file - Which suggests to me the file is damaged in some way. Could you please attach a copy of an SAV file that shows this problem?

**Comment by baltzis** [comment]
Thank you very much for your efforts.

I am attaching a sample file. The same file can be opened without problem
from the route directory, but it can't be opened if you put it in a
folder/subdirectory with Greek characters. The message appears instead.

Regards
Alexandros


2017-03-22 17:01 GMT+02:00 Patrick Knight <notifications@github.com>:

**Comment by aknight1-uva** [comment]
Sorry Alexandros, I can't see your attachment. Did you attach it to an comment on typed-in on  github.com?

**Comment by baltzis** [comment]
Sorry, I've sent it through an email. Here is the file.
[Flexibility scale1.zip](https://github.com/jasp-stats/jasp-desktop/files/862034/Flexibility.scale1.zip)

**Comment by aknight1-uva** [comment]
@baltzis : Thanks for the file. There doesn't appear to be any damage to it at all.
@boutinb, @FransMeerhoff  Further investigation has produced a possible cause of this issue: The importers (without exception) use a std::ifstream object to read files. With underlying files-systems that support UTF-8 filenames (e.g. Linux or ?OS/X?) this works really well in the presence of Unicode chars in the  path/file name. Windows doesn't do this: On Windows, the file system for MBCS (8 bit char) file names reverts to the OEM/code page character set (or did up to Win XP: Vista and latter is anyone's guess, Microsoft seems very reticent on the issue).
In short, the chances of JASP on Windows importing any file with one or more non ASCII character(s) in the path or file name is at best small, at worst vanishingly remote.
The fix would be to replace std::ifstream with the Qt equivalent, assuming it is working properly on Windows builds. Given the age of the Qt libraries used with Windows builds, I would have my doubts about this.

**Comment by baltzis** [comment]
Should I understand this as a final result, saying that there is no solution for Windows in languages other than English?

**Comment by boutinb** [comment]
@baltzis No, there should be a solution. But first of all, we have to reproduce your problem. I have just tried to make folder with a greek's name on a Windows 7 machine, and I can load a JASP file without problem. We will try on a Windowns 10 machine.

**Comment by aknight1-uva** [comment]
All we can say at present, is that we cannot reproduce the issue on Windows 7 (English).
For now, it seems the only work around we can offer is one you already have: Keep all your SAV files you wish to read with JASP in a folder whose path contains only ASCII characters (i.e. only those used by a US keyboard). This is an issue that will require some careful thought - In other words will take some time to fix.

**Comment by baltzis** [comment]
Thank you.

**Comment by aknight1-uva** [comment]
I can reproduce this in Windows 10 (Enterprise, English) using Greek letters in file name.
However the QT Creator required for Windows builds won't run in Win 10 (it GPF's on start-up), even after running qtbinpatcher,exe.

**Comment by baltzis** [comment]
The new version 8.1.2 keeps having the same trouble, but not only with non-ASCII characters in the path name. For some reason, even in a path with only ASCII characters in the path name, I get the following message:
Unable to open file
No header found in .SAV file.

Very annoying, because I would like to use JASP for teaching.

**Comment by baltzis** [comment]
Interestingly enough, when this problem appears with one file once, JASP seems to have problem with any file in the same directory. While at first it opened for e.g. file A, after encountering problem with file B, it does no longer open file A, not any other file in the same directory.

**Comment by baltzis** [comment]
Another interesting detail: the same file can be opened on my 64bit desktop, but on my 32bit laptop. The two computers are connected on a home network. The desktop 64bit computer cannot open the same file from the 32bit laptop when read through the network. The same message appears:
Unable to open file
No header found in .SAV file.

I am not sure whether this additional information helps. I hope it does.
